Mathlib/Topology/Basic.lean

```diff
@@ -1627,7 +1627,10 @@ scoped[Topology] notation (name := Continuous_of) "Continuous[" t₁ ", " t₂ "
 
 variable {f : X → Y} {s : Set X} {x : X} {y : Y}
 
-theorem continuous_def : Continuous f ↔ ∀ s, IsOpen s → IsOpen (f ⁻¹' s) :=
+-- The curly braces are intentional, so this definitions works well with simp
+-- when topologies are not those provided by instances.
+theorem continuous_def {_ : TopologicalSpace X} {_ : TopologicalSpace Y} :
```

Comment:
I don't know whether this is necessarily the right thing to do (my downstream code does not care one way or the other), but I just wanted to note: to fully revert the change, you would need to re-add the `{f : X → Y}` parameter here after the `TopologicalSpace` parameters. Then I think you would not need the other changes in `Order.lean`.
https://github.com/leanprover-community/mathlib4/blob/e7170d39769f41a993b9e30c9eff0f80c3fac97d/Mathlib/Topology/Basic.lean#L1631-L1632

Reply:
Indeed, I think it's better to have `f` come last.

Reply:
Fair point. I have changed the order.